### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by moseshll**
This is not meant to be merged as a single branch. The tests and Perl::Critic changes should be pulled out into separate branches since they are independent. But the Docker stuff made those possible, so I'm getting all the code changes archived in one place for now.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,167 @@
+FROM debian:buster
```
Comment (aelkiss) [inline Dockerfile:1]: You could probably use

`FROM hathitrust/feed-base:buster` and pull in most of this stuff (along with some dependencies you don't need)

Eventually we should make a common base image for HT web apps

File: docker/db/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM mariadb:10.1
```
Comment (aelkiss) [inline docker/db/Dockerfile:1]: Likewise this should probably be a shared image that has the base tables, but this looks good for now (I haven't published anything with the base tables yet anyway)

**Review by aelkiss — APPROVED** [review]
This looks like a great start!

As we manage to dockerize more things we should pull the common functionality out into base images we can re-use across applications, and especially think about how to manage the database schema (probably we have common base tables for e.g. rights and applications that manage other tables/databases add schema files as needed).

Have you looked at adding Travis for CRMS with these tests?

File: Dockerfile (hunk 1 above)
Comment (moseshll) [reply Dockerfile:1]: I started with hathitrust/feed-base:buster and then walked it back to something I had more fine-grained control over. Mostly to get a feel for what was happening behind the scenes. I'll stick with that for now as I get a handle on whether and what we have in terms of weirder dependencies.

Note also that the effort I have thus far doesn't include any webserver. I flailed around with that for a while and then gave up as I have zero experience with Apache/nginx/whatever config.

I may write a ticket for Travis-izing the tests. What is there is just barely enough for a go/no-go, and it probably would have saved my butt a couple times in the last decade if I had had such.

Comment (aelkiss) [reply Dockerfile:1]: Sounds good. I'm interested in getting something that can replicate development end-to-end on a local workstation, including something more robust to impersonate Shibboleth identities in development. I'm not sure whether it makes sense to fold that in with some of the download module work or put it forth as a separate project in half-cycle planning, but either way it's something I'd like to move forward with.

**Pull request closed without merging**